### PR TITLE
Refactor the list collector repeating blocks functional test spec

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?
Refactored the tests in the list collector repeating blocks spec file. Needed to be refactored as some of the test cases were missing assertions or were executing additional steps after assertions. All tests should now be following the Given, When, Then pattern.

### How to review
Check that the changes make sense and that the functional test still passes. Are any test cases missing?

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by berroar** [comment]
[Refactor the list_collector_repeating_blocks spec (1 SP)](https://trello.com/c/lRUrVofO/5842-refactor-the-listcollectorrepeatingblocks-spec-1-sp)

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js [hunk 1]
```diff
@@ -52,13 +53,13 @@ async function addCompany(
 }
 
 describe("List Collector Repeating Blocks", () => {
-  describe("Given a normal journey through the list collector with repeating blocks, the answers can be submitted.", () => {
+  describe("Given a normal journey through the list collector with repeating blocks", () => {
```
Comment (katie-gardner) [inline tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:56]: Minor - whats the convention for `describe` naming?
I find it a bit strange that the describe is `Given a normal journey.` without any kind of when, then
I'd favour changing it to just `describe('A normal journey...')`

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js [hunk 2]
```diff
@@ -71,14 +72,15 @@ describe("List Collector Repeating Blocks", () => {
       await click(AnyOtherTradingDetailsPage.submit());
       await click(SectionCompaniesPage.submit());
       await click(SubmitPage.submit());
+      await expect(await browser.getUrl()).toContain(ThankYouPage.pageName);
     });
   });
 
-  describe("Given a journey through the list collector with repeating blocks, the companies can be added, removed and edited.", () => {
+  describe("Given a journey through the list collector with repeating blocks", () => {
```
Comment (katie-gardner) [inline tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:79]: Same comment - would we want this to be something like `A journey with list edits through a list collector with repeating blocks` to give a bit of context but not overdo it?

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js [hunk 3]
```diff
@@ -87,57 +89,48 @@ describe("List Collector Repeating Blocks", () => {
...
       await click(AnyOtherCompaniesOrBranchesPage.submit());
 
       await click(AnyOtherTradingDetailsPage.submit());
       await click(SectionCompaniesPage.submit());
       await click(SubmitPage.submit());
+      await expect(await browser.getUrl()).toContain(ThankYouPage.pageName);
     });
   });
 
   describe("Given a journey through the list collector with repeating blocks, any incomplete repeating_block will be revisited before navigating to the next block after the list collector.", () => {
```
Comment (katie-gardner) [inline tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:129]: `Given a journey that tests routing through....` ? Totally open to other thoughts, but I reckon these could be a bit more consistent, the other two are very generic and this one is quite specific.

**Review by katie-gardner — COMMENTED** [review]
This is way better, nice 👍 
Changes look good, includes all the correct tests with proper assertions
Only minor thought was that the describe headers were slightly inconsistent, some quite detailed and some not, reckon middle ground for those would be nice

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js (hunk 1 above)
Comment (katie-gardner) [reply tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:56]: ahhh, I see, that makes more sense to me now - the latter ones with Given a journey..., any incomplete... don't fit this style but I think that works if they're made consistent with these ones

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js (hunk 2 above)
Comment (berroar) [reply tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js]: Amended slightly now 👍

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js (hunk 3 above)
Comment (berroar) [reply tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js]: Amended slightly now 👍

**Review by katie-gardner — APPROVED** [review]

**Review by Yuyuutsu — APPROVED** [review]
Changes look good and working as expected 👍

File: tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js [hunk 4]
```diff
@@ -71,14 +72,15 @@ describe("List Collector Repeating Blocks", () => {
       await click(AnyOtherTradingDetailsPage.submit());
       await click(SectionCompaniesPage.submit());
       await click(SubmitPage.submit());
+      await expect(await browser.getUrl()).toContain(ThankYouPage.pageName);
```
Comment (VirajP1002) [inline tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:75]: Would it be better for the test to be `await expect(browser).toHaveUrlContaining(ThankYouPage.pageName);` to match the new formatting of the tests?

Comment (berroar) [reply tests/functional/spec/features/repeating_blocks/list_collector_repeating_blocks.spec.js:75]: Was awaiting merge of your PR and the inevitable merge conflicts before switching to use those 👍

**Pull request merged**